prometheus/testutil/testutil.go

```diff
@@ -108,6 +108,34 @@ func ToFloat64(c prometheus.Collector) float64 {
 	panic(fmt.Errorf("collected a non-gauge/counter/untyped metric: %s", pb))
 }
 
+// MetricCount collects all Metrics from the provided Collector and counts them.
```

Comment:
Suggestion: MetricCount collects all Metrics from the provided Collector and returns their number.